security-framework/src/os/macos/passwords.rs

```diff
@@ -0,0 +1,166 @@
+//! Password support.
+
+use security_framework_sys::passwords::*;
+use security_framework_sys::base::errSecSuccess;
+use security_framework_sys::keychain_item::{SecKeychainItemDelete,
+                                            SecKeychainItemModifyAttributesAndData};
+use core_foundation_sys::base::CFRelease;
+use std::ptr;
+use std::ffi::CString;
+use libc::c_void;
+
+use cvt;
+use base::Result;
+
+/// Find a generic password in the default list of keychains.
+///
+/// The underlying system supports passwords with 0 values, so this
+/// returns a vector of bytes rather than a string.
+pub fn find_generic_password(service: &str, account: &str) -> Result<Vec<u8>> {
```

Comment:
It seems like these should be made a bit more powerful to cover the functionality the underlying function provides. In particular, the optional specification of a keychain list and the option to retrieve the keychain item.